### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by sonniesedge**
Travis is [no longer a great place for Open Source software](https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works), so we're looking to move to another platform for running CI tasks.

This PR moves the CI from Travis to Github Actions.

## Contribution note
 This PR uses the wonderful commits supplied by @paazmaya in #548.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Changelog
 
+* Migrate away from `make` commands in testing
```
Comment (joeyciechanowicz) [inline CHANGELOG.md:3]: 🎉

**Review by joeyciechanowicz — APPROVED** [review]
Oh oh oh! Yes please!

**Comment by joeyciechanowicz** [comment]
image

I'm guessing we need to remove the travice integration in the settings before we can merge this PR then.

File: CHANGELOG.md (hunk 1 above)
Comment (sonniesedge) [reply CHANGELOG.md:3]: Joey, could you pls take a look at the (very welcome and excellent!) commits by @paazmaya? The ones that remove make and substitute npm?I think all the tasks from @rowanmanning's [makefile](https://github.com/rowanmanning/make/blob/master/javascript/index.mk) are there but I'm not 100% sure as I'm JUST SO NEW to the codebase.

**Comment by sonniesedge** [comment]
> image
> 
> I'm guessing we need to remove the travice integration in the settings before we can merge this PR then.

Yeah, and we'll also have to have a followup PR to remove the current PR branch from the list of branches taht are tested.

File: CHANGELOG.md (hunk 1 above)
Comment (joeyciechanowicz) [reply CHANGELOG.md:3]: Yup, manually checked and @paazmaya has replicated everything the `make ci` command was doing ⭐ .

**Comment by joeyciechanowicz** [comment]
> Just to note that a couple of times during development the tests time out, necessitating a rerun of the job to pass.
> 
> Were the tests mildly flaky when using Travis?

I seem to recall that when I did a migration to `jest` (and lost the branch 😭 ) there where some tests which weren't properly cleaning up promises, so executing order could cause flakiness. 
And on Travis, I do recall re-running builds sometimes.

**Comment by hollsk** [comment]
Yeah now that you mention it, I also remember builds on Travis occasionally flaking out for no obvious reason.

File: CHANGELOG.md (hunk 1 above)
Comment (sonniesedge) [reply CHANGELOG.md:3]: Wonderful!

**Comment by sonniesedge** [comment]
So, shall we make the transfer to Github Actions now, before the weekend gets close? 

Someone with admin rights on this repo will need to disable Travis integration before this PR can be merged.

**Comment by sonniesedge** [comment]
BTW, again as a new contributor, do we have other branches that are actively maintained? If so they will need this PR backporting to them to enable Github Actions to test them.

**Comment by hollsk** [comment]
GOOD QUESTION! I had a look at all the branches that are open, and I see that Joey has been a branch open for merging in the reporters, the exact thing we've been talking about doing this week! 🎉 

Not sure what Jose's test branch is for, but none of the others are actively maintained. 

I'm definitely voting for "sooner" rather than "later" on the transfer front 👍

**Comment by joeyciechanowicz** [comment]
We only build main and PR branches, all the other branches don't need to be build IMO.

**Comment by sonniesedge** [comment]
> We only build main and PR branches, all the other branches don't need to be build IMO.

Okay, but we'd still need to build and test PRs against older branches, no? Say if there was a security issue.

We could handle that by adding the main version branches [to the PR array](https://github.com/pa11y/pa11y/pull/551/files#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fR10).

**Comment by joeyciechanowicz** [comment]
> Say if there was a security issue for 4.x?

We don't get security issues .... 👀 

> adding the main version branches to the PR array

Great idea. That works for me!

**Comment by josebolos** [comment]
> BTW, again as a new contributor, do we have other branches that are actively maintained? If so they will need this PR backporting to them to enable Github Actions to test them.

Good thinking, we do, 5.x and 5.2.x are the current stable release at the moment. `master` is the 6.0 alpha.

**Comment by paazmaya** [comment]
Could this configuration be copied across all Pa11y repositories?

**Pull request merged**